haskell/private/dependencies.bzl

```diff
@@ -37,15 +37,16 @@ def gather_dep_info(ctx, deps):
         if HaskellInfo in dep
     ])
 
-    source_files = set.empty()
-    for dep in deps:
-        if HaskellInfo in dep:
-            source_files = set.mutable_union(source_files, dep[HaskellInfo].source_files)
+    source_files = depset(transitive = [
+        dep[HaskellInfo].source_files
+        for dep in deps
+        if HaskellInfo in dep
+    ])
 
     import_dirs = set.empty()
     for dep in deps:
         if HaskellInfo in dep:
-            import_dirs = set.mutable_union(source_files, dep[HaskellInfo].source_files)
+            import_dirs = set.mutable_union(import_dirs, dep[HaskellInfo].import_dirs)
```

Comment:
Looks like a copy-paste issue. Interesting that this didn't cause any problems.

Reply:
Yes. The only place where `import_dirs` are taken from a provider is in the repl code. I don't understand `import_dirs` well enough yet to know whether the repl code really needs them. My suspicion is that `import_dirs` is unnecessary altogether.